packages/administration/src/Controller/DashboardController.php

```diff
@@ -0,0 +1,450 @@
+            'action' => $this->generateUrl('admin_product_list'),
+        ]);
+
+        $currentCountOfOrders = $this->statisticsFacade->getOrdersCount(static::PREVIOUS_DAYS_TO_LOAD_STATISTICS_FOR);
+        $previousCountOfOrders = $this->statisticsFacade->getOrdersCount(
+            static::PREVIOUS_DAYS_TO_LOAD_STATISTICS_FOR * 2,
+            static::PREVIOUS_DAYS_TO_LOAD_STATISTICS_FOR,
+        );
+
+        $ordersTrend = $this->getTrendDifference($previousCountOfOrders, $currentCountOfOrders);
+
+        $currentCountOfNewCustomers = $this->statisticsFacade->getNewCustomersCount(
+            static::PREVIOUS_DAYS_TO_LOAD_STATISTICS_FOR,
+        );
+        $previousCountOfNewCustomers = $this->statisticsFacade->getNewCustomersCount(
+            static::PREVIOUS_DAYS_TO_LOAD_STATISTICS_FOR * 2,
+            static::PREVIOUS_DAYS_TO_LOAD_STATISTICS_FOR,
+        );
+
+        $newCustomersTrend = $this->getTrendDifference($previousCountOfNewCustomers, $currentCountOfNewCustomers);
+
+        $currentValueOfOrders = $this->statisticsFacade->getOrdersValue(static::PREVIOUS_DAYS_TO_LOAD_STATISTICS_FOR);
+        $previousValueOfOrders = $this->statisticsFacade->getOrdersValue(
+            static::PREVIOUS_DAYS_TO_LOAD_STATISTICS_FOR * 2,
+            static::PREVIOUS_DAYS_TO_LOAD_STATISTICS_FOR,
+        );
+
+        $ordersValueTrend = $this->getTrendDifference($previousValueOfOrders, $currentValueOfOrders);
+
+        $this->addWarningMessagesOnDashboard();
+
+        return $this->render(
+            '@ShopsysAdministration/dashboard/dashboard.html.twig',
+            [
+                'registeredInLastTwoWeeksLabels' => $registeredInLastTwoWeeksDates,
+                'registeredInLastTwoWeeksValues' => $registeredInLastTwoWeeksCounts,
+                'newOrdersInLastTwoWeeksLabels' => $newOrdersInLastTwoWeeksDates,
+                'newOrdersInLastTwoWeeksValues' => $newOrdersInLastTwoWeeksCounts,
+                'quickProductSearchForm' => $quickProductSearchForm->createView(),
+                'newCustomers' => $currentCountOfNewCustomers,
+                'newCustomersTrend' => $newCustomersTrend,
+                'newOrders' => $currentCountOfOrders,
+                'newOrdersTrend' => $ordersTrend,
+                'ordersValue' => $currentValueOfOrders,
+                'ordersValueTrend' => $ordersValueTrend,
+                'transferIssuesCount' => $transferIssuesCount,
+                'cronGridViews' => $this->getCronGridViews(),
+            ],
+        );
+    }
+
+    /**
+     * @return \Shopsys\FrameworkBundle\Model\Administrator\Administrator
+     */
+    protected function getCurrentAdministrator(): Administrator
+    {
+        /** @var \Shopsys\FrameworkBundle\Model\Administrator\Administrator|null $administrator */
+        $administrator = $this->getUser();
+
+        if ($administrator === null) {
+            throw $this->createAccessDeniedException('No administrator is logged in.');
+        }
```

Comment:
You should be able to do this check in security.yaml, right?

Reply:
this code is copy-paste from the current admin. Basically to ensure the admin is not null as controller's `$this->getUser();` may return null.